### ISSUE TITLE
Unable to install metaocaml

### ISSUE THREAD

**Issue by abhi18av** [yallop/metaocaml-letrec#3]
Hi @yallop 

I am trying to install the `metaocaml` variant to try out this package but opam isn't responding as per the `ReadMe.md` instructions.

I reached this repo via the following paper https://www.cl.cam.ac.uk/~jdy22/papers/generating-mutually-recursive-definitions.pdf 

First off, I'm only able to run `opam` when using `sudo` as mentioned here https://github.com/ocaml/opam-repository/issues/13111#issuecomment-445483789 

```sh
Abhinavs-MacBook-Pro:~ eklavya$ sudo opam switch 4.07.1+BER
[WARNING] Running as root is not recommended
# Run eval $(opam env) to update the current shell environment
Abhinavs-MacBook-Pro:~ eklavya$ eval $(opam config env)
Abhinavs-MacBook-Pro:~ eklavya$ eval $(opam env)
Abhinavs-MacBook-Pro:~ eklavya$ ocaml
bash: ocaml: command not found
Abhinavs-MacBook-Pro:~ eklavya$ opam env
OPAM_SWITCH_PREFIX='/Users/eklavya/.opam/4.07.1+BER'; export OPAM_SWITCH_PREFIX;
```

How can I make all of this work?

**Comment by abhi18av** [comment]
Even after that, I run into the following error 

```sh

Abhinavs-MacBook-Pro:~ eklavya$ sudo opam switch create 4.07.1+BER
[WARNING] Running as root is not recommended

<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫
[ocaml-variants.4.07.1+BER] found in cache

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫
∗ installed base-bigarray.base
∗ installed base-threads.base
∗ installed base-unix.base
[ERROR] The compilation of ocaml-variants failed at "/Users/eklavya/.opam/opam-init/hooks/sandbox.sh build ./configure -prefix
        /Users/eklavya/.opam/4.07.1+BER -cc cc -aspp cc -c".

#=== ERROR while compiling ocaml-variants.4.07.1+BER ==========================#
# context     2.0.1 | macos/x86_64 |  | https://opam.ocaml.org#41ad4464
# path        ~/.opam/4.07.1+BER/.opam-switch/build/ocaml-variants.4.07.1+BER
# command     ~/.opam/opam-init/hooks/sandbox.sh build ./configure -prefix /Users/eklavya/.opam/4.07.1+BER -cc cc -aspp cc -c
# exit-code   2
# env-file    ~/.opam/log/ocaml-variants-54088-ffb3fd.env
# output-file ~/.opam/log/ocaml-variants-54088-ffb3fd.out
### output ###
# Configuring OCaml version 4.07.1
# Configuring for host x86_64-apple-darwin17.5.0 ...
# Configuring for target x86_64-apple-darwin17.5.0 ...
# Using compiler cc.
# cc: error: couldn't create cache file '/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/xcrun_db-0I43LQsf' (errno=Operation not permitted)
# cc: error: couldn't create cache file '/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/xcrun_db-nEKsLkkR' (errno=Operation not permitted)
# cc: error: couldn't create cache file '/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/xcrun_db-EIDDLBrr' (errno=Operation not permitted)
# Compiler family and version: clang-9-1.
# [ERROR!] Unable to compile the test program.
#  Make sure the C compiler cc is properly installed.



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫
┌─ The following actions failed
│ λ build ocaml-variants 4.07.1+BER
└─
┌─ The following changes have been performed (the rest was aborted)
│ ∗ install base-bigarray base
│ ∗ install base-threads  base
│ ∗ install base-unix     base
└─
# Run eval $(opam env) to update the current shell environment
Switch initialisation failed: clean up? ('n' will leave the switch partially installed) [Y/n]
```

I have already done 

```
sudo xcode-select --reset
```

Please guide me a bit here :)

**Comment by abhi18av** [comment]
I have since tried working with the environment variables like 

```

alias gcc='gcc-8'
alias cc='gcc-8'
alias g++='g++-8'
alias c++='c++-8'


```

But when I try to install `metaocaml`, it still seems to be using `clang`

```
Abhinavs-MacBook-Pro:~ eklavya$ sudo opam switch create 4.07.1+BER
[WARNING] Running as root is not recommended

<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫
[ocaml-variants.4.07.1+BER] found in cache

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫
∗ installed base-bigarray.base
∗ installed base-threads.base
∗ installed base-unix.base
[ERROR] The compilation of ocaml-variants failed at "/Users/eklavya/.opam/opam-init/hooks/sandbox.sh build ./configure -prefix
        /Users/eklavya/.opam/4.07.1+BER -cc cc -aspp cc -c".

#=== ERROR while compiling ocaml-variants.4.07.1+BER ==========================#
# context     2.0.1 | macos/x86_64 |  | https://opam.ocaml.org#41ad4464
# path        ~/.opam/4.07.1+BER/.opam-switch/build/ocaml-variants.4.07.1+BER
# command     ~/.opam/opam-init/hooks/sandbox.sh build ./configure -prefix /Users/eklavya/.opam/4.07.1+BER -cc cc -aspp cc -c
# exit-code   2
# env-file    ~/.opam/log/ocaml-variants-56546-ffb3fd.env
# output-file ~/.opam/log/ocaml-variants-56546-ffb3fd.out
### output ###
# Configuring OCaml version 4.07.1
# Configuring for host x86_64-apple-darwin17.5.0 ...
# Configuring for target x86_64-apple-darwin17.5.0 ...
# Using compiler cc.
# cc: error: couldn't create cache file '/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/xcrun_db-SLFHwPXO' (errno=Operation not permitted)
# cc: error: couldn't create cache file '/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/xcrun_db-nUGItVQz' (errno=Operation not permitted)
# cc: error: couldn't create cache file '/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/xcrun_db-gD5ZIuIZ' (errno=Operation not permitted)
# Compiler family and version: clang-9-1.
# [ERROR!] Unable to compile the test program.
#  Make sure the C compiler cc is properly installed.



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫
┌─ The following actions failed
│ λ build ocaml-variants 4.07.1+BER
└─
┌─ The following changes have been performed (the rest was aborted)
│ ∗ install base-bigarray base
│ ∗ install base-threads  base
│ ∗ install base-unix     base
└─
# Run eval $(opam env) to update the current shell environment
Switch initialisation failed: clean up? ('n' will leave the switch partially installed) [Y/n] n
Abhinavs-MacBook-Pro:~ eklavya$ cc
gcc-8: fatal error: no input files
compilation terminated.
```

**Comment by abhi18av** [comment]
Hi @yallop , I have realized the issue now.

Initially I had installed `opam` via `homebrew` which resulted in `opam` not being able to `update` or `install` anything.

Then eventually, after trying a lot many things, I just did the vanilla `shell script` based installation as mentioned in the website and then everything started working fine.

I've since been able to install and setup `4.07.1+BER` without any issues 👍
